### DISCUSSION TITLE
Virtualization-Issue Windows Server 2019

### DISCUSSION THREAD

**Discussion by AndreasMAlz** [jamulussoftware/jamulus#1462] [General]
**Describe the bug**
Terrible Audio-Performance von Windows Server 2019 at HyperV (Strato V-Server V40).

**To Reproduce**
Connect to private Jamulus-Server on Strato V-Server Windows V40.

**Expected behavior**
Perfect Audio-Quality at a Server with 6 vCores, 8GB RAM and 1 Gbit/s network rate.

**Screenshots**
It's just to hear.

**Operating system**
Clients: Windows 7, 8 and 10, MacOS (all clients have approx. the same Audio-Issues)
Server: Windows Server 2019

**Version of Jamulus**
3.7.0 on Server and some clients. Other clients still use 3.6.2

**Additional context**
I carried out some rehearsals of a choir with up to 30 singers, equipped with good headsets, at a virtual machine on Microsoft Azure (Standard B2ms, Region Germany-West-Central). The audio-quality was quiet good (Settings: mono-in, stereo-out, "normal" quality). After change over to a V-Server hosted by Strato.de (HyperV) we have growing issues with the sound quality. While ping-times are good, the singers complains "bubble-sound" and drop outs. Latency and Puffer are mostly red or yellow, rarely green. The amount of virtual RAM on the HyperV-Server from Strato ist the same as at MS Azure. The Azure-VM had only 2 vCPUs but 50 GBit/s network-rate, while the Strato-Server just have "up to 1 GBit/s" connection. I guess the reason is the virtualization, that provide just the amount of RAM that actually is needed by Windows. Or are there other possible causes? How should I configure the virtual Windows Server 2019 to provide optimal performance to Jamulus? Thanks beforhand.

**Comment by hoffie** [comment]
Thanks for your report. We're using the Github issues for bugs and fully specified feature requests only while your question sounds like a support request. I'm therefore moving this to a discussion. :)

**Comment by hoffie** [comment]
Jamulus' RAM usage should be stable (once all clients have connected), so I don't think this is the issue.
I rather suspect that your VM might not have a dedicated CPU core but might share it with its neighbours on the same virtualization host.
Jamulus needs low-latency, predictable CPU time.

Network may also be a reason. If packets are not forwarded to your VM fast enough (either due to routing or hypervisor/VM communication), Jamulus will not get a chance to provide fast enough handling.

Sadly, I'm neither a HyperV, Windows or Strato specialist, so I'm hoping others will chime in with ideas how to debug and improve further. :)

**Comment by dcorson-ticino-com** [comment]
Bubble sound is more often related to network problems, not the server itself.
It appears when the packets arrive out of order.
Jamulus already has implemented packet ordering, but the amount that can be corrected is not that big to not increase the delay times.
That packet ordering already made a big improvement of the "bubbling" problem, but has limits.
I would suggest that you talk to your provider if you can specify the quality of the network connection you are getting.
Network speed is not the only factor.

**Comment by AndreasMAlz** [comment]
Thank you for your replies. I wrote this points to the Strato-Support and hope for an answer, what can be done to improve that.

In Windows in the properties for the Network-Adapter there are some possible settings, such as Puffer-Size, Forwarding Optimization and so on. Have anyone experiences with this?

![Bildschirmfoto 2021-04-08 um 16 15 39](https://user-images.githubusercontent.com/82094137/114042630-e1b8f780-9885-11eb-83ef-7483f78a1e32.png)

**Reply by dcorson-ticino-com** [reply]
A long time ago I tried to do some optimization at that level for a client, but I never saw any difference in the performance and reset all to default.
